atom/renderer/extensions/atom_extensions_renderer_client.cc

```diff
@@ -0,0 +1,58 @@
+// Copyright 2014 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "atom/renderer/extensions/atom_extensions_renderer_client.h"
+
+#include "content/public/renderer/render_thread.h"
+#include "extensions/renderer/dispatcher.h"
+#include "extensions/renderer/dispatcher_delegate.h"
+
+namespace atom {
+
+AtomExtensionsRendererClient::AtomExtensionsRendererClient()
+    : dispatcher_(std::make_unique<extensions::Dispatcher>(
+          std::make_unique<extensions::DispatcherDelegate>())) {
+  dispatcher_->OnRenderThreadStarted(content::RenderThread::Get());
+}
+
+AtomExtensionsRendererClient::~AtomExtensionsRendererClient() {}
+
+bool AtomExtensionsRendererClient::IsIncognitoProcess() const {
+  // app_shell doesn't support off-the-record contexts.
+  return false;
+}
+
+int AtomExtensionsRendererClient::GetLowestIsolatedWorldId() const {
+  // app_shell doesn't need to reserve world IDs for anything other than
+  // extensions, so we always return 1. Note that 0 is reserved for the global
+  // world.
+  // TODO(samuelmaddock): skip electron worlds
+  return 10;
```

Comment:
:thinking: 

This should be calculated to skip the isolated world rather than a hardcoded magic number

Reply:
We'll want to use [`World::ISOLATED_WORLD_EXTENSIONS`](https://github.com/electron/electron/blob/53f6cbccbf388dd28bd8636cd452df4c1d0f4f13/atom/renderer/atom_render_frame_observer.h#L33)